dts/bindings/mtd/cypress,fm25w256.yaml

```diff
@@ -0,0 +1,25 @@
+# Copyright (c) 2021 Caspar Friedrich <c.s.w.friedrich@gmail.com>
+# SPDX-License-Identifier: Apache-2.0
+
+description: 256-Kbit (32K × 8) Serial (SPI) F-RAM
+
+compatible: "cypress,fm25w256"
+
+include: [eeprom-base.yaml, spi-device.yaml]
+
+properties:
+  wp-gpios:
+    type: phandle-array
+    required: false
+    description: Enable write protection on status register. Active low.
+
+  hold-gpios:
+    type: phandle-array
+    required: false
+    description: Suspend the current spi operation. Active low.
+
+  spi-mode-3:
+    type: boolean
+    description: |
+      Enable SPI Mode 3 (CPOL = 1, CPHA = 1) for this device. If unset, device
+      operates in SPI Mode 0 (CPOL = 0, CPHA = 0).
```

Comment:
@galak Should we consider a common SPI device property (enum) for setting the SPI mode for devices supporting different modes?

Reply:
Done in the driver for now.